### PR TITLE
Search Options Accessibility

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#15031

*Description of changes:*
* Minor Role/Product refactors for consistency between client and staff accounts.
* Additional error handling/control on MHR only accessibility.
* Hide Registration table for MHR product only client accounts.
* Minor Template/styling updates
* Minor code clean up

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by ChasBoucher — APPROVED** [review]
Looks good..

**Review by doug-lovett — APPROVED** [review]
Looks good.

File: ppr-ui/src/App.vue [hunk 1]
```diff
@@ -687,7 +703,7 @@ export default class App extends Mixins(AuthMixin) {
   private proceedAfterError (proceed: boolean): void {
     this.errorDisplay = false
     // still need to fill this out more
-    if (this.errorOptions === loginError || this.errorOptions === authError) {
+    if (this.errorOptions === loginError || this.errorOptions === authError || this.errorOptions === authAssetsError) {
```
Comment (dimak1) [inline ppr-ui/src/App.vue]: A little bit confusing with `authError` vs `authAssetsError`.

File: ppr-ui/src/components/search/SearchBarList.vue [hunk 2]
```diff
@@ -106,44 +109,45 @@ export default defineComponent({
...
         }
-        if (hasPprRole.value) {
-          allSearchTypes.push.apply(allSearchTypes, SearchTypes)
-          // we can pop the title off if there is only one search type
-          if (!hasMhrRoleEnabled.value) {
-            allSearchTypes.shift()
-          }
+
+        // Client Only Blocks
+        if (hasPprEnabled.value && hasMhrEnabled.value) {
```
Comment (dimak1) [inline ppr-ui/src/components/search/SearchBarList.vue:125]: Nice refactor 👍🏻

File: ppr-ui/src/components/search/SearchBarList.vue [hunk 3]
```diff
@@ -106,44 +109,45 @@ export default defineComponent({
...
+      isSingleSearchOption: computed((): boolean => {
+        return (hasPprEnabled.value && !hasMhrEnabled.value) || (!hasPprEnabled.value && hasMhrEnabled.value)
       }),
-      isPPROnly: computed((): boolean => hasPprRole.value && !hasMhrRoleEnabled.value),
       displayItems: [],
       displayGroup: {
-        1: true, // default, fallback state for safety
-        2: false
+        1: hasPprEnabled.value && !hasMhrEnabled.value && !isRoleStaff.value,
+        2: !hasPprEnabled.value && hasMhrEnabled.value && !isRoleStaff.value
```
Comment (dimak1) [inline ppr-ui/src/components/search/SearchBarList.vue]: This should be open when there is only MHR, right?

File: ppr-ui/src/App.vue [hunk 4]
```diff
@@ -358,6 +360,17 @@ export default class App extends Mixins(AuthMixin) {
       }
     }
 
+    // Safety check for client account products
```
Comment (dimak1) [inline ppr-ui/src/App.vue:363]: Nice to see a fallback scenario(s), to help prevent clients reporting app issues 👍🏻

**Review by dimak1 — APPROVED** [review]
All good, approved. Some clarification comments.

File: ppr-ui/src/components/search/SearchBarList.vue (hunk 3 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/search/SearchBarList.vue]: Yessir, it is indeed.

File: ppr-ui/src/App.vue (hunk 4 above)
Comment (cameron-eyds) [reply ppr-ui/src/App.vue:363]: Definitely. 
Helps in the scenario you were thinking about previously, where perhaps our feature flag to enable MHR is out of sync with Auths Mhr feature flag. 
Shouldn't happen " " , but at least there is some protection.

File: ppr-ui/src/App.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/App.vue]: Hmm yes, i'll have another look. 
Maybe the name could be better!

Comment (cameron-eyds) [reply ppr-ui/src/App.vue]: Actually updated `authError` to `authPprError` to be more concise, since it is handling `PPR` access errors explicitly.

Comment (dimak1) [reply ppr-ui/src/App.vue]: Great, thanks, yea i think it's better.

**Review by ChasBoucher — APPROVED** [review]
Looks Good

**Review by ChasBoucher — COMMENTED** [review]
Looks good

**Review by ChasBoucher — APPROVED** [review]
Looks good..

**Pull request merged**